### ISSUE TITLE
[$250] When a receipt matches with a personal card transaction, it still shows Amount • Cash above the amount

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#46554]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.14-5
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1722282008985029

## Action Performed:

1. Submit a card expense report with receipt
2. Open the report as approver (This is report 6565029153383101)

## Expected Result:
The expense is mentioned as card expense

## Actual Result:
Shown as Amount • Cash above the amount
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Screenshot 2024-07-29 at 3 38 34 PM](https://github.com/user-attachments/assets/42d36d5f-469a-44fd-891a-dfe49be5e5ce)

![Screenshot 2024-07-29 at 3 38 51 PM](https://github.com/user-attachments/assets/f4ba1dd4-7af2-4003-9dee-82cded2a7ca7)

![Snip - (518) New Expensify - Google Chrome](https://github.com/user-attachments/assets/108a8cca-baba-4021-b16f-586503d618f6)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0184082e81b7b23b49</li>
        <li>Upwork Job ID: 1820474809466444194</li>
        <li>Last Price Increase: 2025-01-20</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Ollyws</details>

**Comment by Chefkj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Amount type is cash when receipt is processed.
### What is the root cause of that problem?
missing logic leads to only cash as an option.
### What changes do you think we should make in order to solve the problem?
update logic to add a boolean path to card when receipt is processed.
Result:
<img width="677" alt="Screenshot 2024-08-08 at 6 14 37 PM" src="https://github.com/user-attachments/assets/f6fb00b8-e6d3-4ef8-8b91-1b081dddab35">
<img width="353" alt="Screenshot 2024-08-08 at 6 15 15 PM" src="https://github.com/user-attachments/assets/3f56f8a7-e0d1-4b14-b80a-663c995c792f">

### What alternative solutions did you explore? (Optional)
OCR support for type definitions, couldn't find the OCR engine to test the return definitions, spent a good while trying to find it, but determined it was a suicide mission and out of scope of the ticket.

**Comment by Chefkj** [comment]
I'm submitting an expense with the + button list, "submit expense" per the error video above.

**Comment by Chefkj** [comment]
No problem,  I just went off the pictures.

click FAB
click submit expense
click scan
add receipt
choose workspace
submit expense
after receipt is scanned, amount still shows cash if it's not a corporate card

**Comment by Chefkj** [comment]
Additionally, when you go to search and expenses, the svg graphic has not updated to the card svg

**Comment by Chefkj** [comment]
I've been looking into it pretty heavily, and I don't believe I can access the receipt scanning software to add the card/cash regex to the text output, as it's probably setup through aws or azure or gcp.  Alternately I could install a node package to handle the image to text conversion, but I don't know if that's allowed, this would be my first ticket, I'm really just trying to get a push up so I can apply for permanent open source positions.

**Comment by puneetlath** [comment]
>@puneetlath Any chance you could explain the reproduction steps for this flow? I'm not aquainted with it and the repro steps aren't very clear. Thanks!

1. Import a personal card (this can be done on OldDot) 
2. Scan a receipt for an expense that was incurred on the imported card
3. Wait for the card expense and the receipt to merge

**Comment by Chefkj** [comment]
@Ollyws Sorry I added further information in a comment, I updated the proposal with the information from the comment I made regarding the solution to the issue.

**Comment by Chefkj** [comment]
Then this should work, is there a test card I can use to test it? or do I need to add my card and find a receipt?
```typescript
 if (isCardTransaction || undefined ? transaction?.cardID != null: true) {
        if (formattedOriginalAmount) {
            amountDescription += ` ${CONST.DOT_SEPARATOR} ${translate('iou.original')} ${formattedOriginalAmount}`;
        }
        if (isCancelled) {
            amountDescription += ` ${CONST.DOT_SEPARATOR} ${translate('iou.canceled')}`;
        }
```

**Comment by abekkala** [comment]
@Ollyws can you check the comment from @Chefkj above?

**Comment by Chefkj** [comment]
This is my first ticket, I submitted a request for slack channel access but still haven't received any response

**Comment by abekkala** [comment]
@Ollyws are you still waiting for testing access?

**Comment by Chefkj** [comment]
@Ollyws My kiddo just got a gift card for his birthday and spent it on pokemon cards I have a receipt for, do you think that'd work?

**Comment by abekkala** [comment]
@Ollyws do you have a test card for this one?

**Comment by abekkala** [comment]
@OfstadC 
This one is old, but we're waiting on some cards for testing. Slack thread: https://expensify.slack.com/archives/C02NK2DQWUX/p1728655992233659?thread_ts=1724811067.380789&cid=C02NK2DQWUX

At that point, we'll be able to review/select a viable proposal for fix.

**Comment by abekkala** [comment]
I'm back from ooo - unassigning @OfstadC

**Comment by abekkala** [comment]
https://expensify.slack.com/archives/C02NK2DQWUX/p1729211626012749?thread_ts=1724811067.380789&cid=C02NK2DQWUX

**Comment by abekkala** [comment]
I'm not sure we still have a method of testing personal cards - I'll dig into this one more today...

**Comment by abekkala** [comment]
@mallenexpensify I _think_ this is still occurring but is that due to the card being connected in Classic and then smart scanning in ND? 
Although they match the Card/Cash label isn't able to update?

**Comment by mallenexpensify** [comment]
I threw the `retest weekly` label on here cuz it's 6 months old.  If QA (or anyone else) is still able to reproduce, we should double the price.  

I like you're line of thinking, I've been testing on the user (not admin) side and I'm unsure how clear we are in the report view with card + cash transactions and with them merging (or not).  Also unsure if this is being discussed anywhere
